app/controllers/courses/vacancies_controller.rb

```diff
@@ -33,15 +32,11 @@ def update
   private
 
     def build_course
-      @course = Course.where(provider_code: params[:provider_code]).find(params[:code])
+      @course = Course.where(provider_code: params[:provider_code]).find(params[:code]).first
     end
 
     def build_site_statuses
-      @site_statuses = @course.map(&:site_statuses).flatten
-    end
-
-    def build_first_course
-      @course = @course.first
+      @site_statuses = @course.site_statuses.select { |site_status| site_status.status == 'running' }
```

Comment:
You might want to put the following helper on `SiteStatus`:

```ruby
def running?
  status == 'running'
end
```

and then this can be prettified to

```ruby
@site_statuses = @course.site_statuses.select(&:running?)
```

Reply:
Great shout. Done